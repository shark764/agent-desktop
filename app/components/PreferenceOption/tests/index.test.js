/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import { PreferenceOption } from '..';

describe('<PreferenceOption />', () => {
  const mockSetPreferenceSelected = jest.fn();
  const rendered = shallow(
    <PreferenceOption
      preference="metrics"
      label={{
        id: 'app.containers.AgentPreferencesMenu.metrics',
        defaultMessage: 'Metrics',
      }}
      setPreferenceSelected={mockSetPreferenceSelected}
    />
  );
  it('renders correctly', () => {
    expect(rendered).toMatchSnapshot();
  });
  it('calls setPreferenceSelected when clicked', () => {
    rendered.find('#metrics-preference').simulate('click');
    expect(mockSetPreferenceSelected).toBeCalledWith('metrics');
  });
});
