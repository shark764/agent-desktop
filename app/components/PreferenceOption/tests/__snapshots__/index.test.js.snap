// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreferenceOption /> renders correctly 1`] = `
<styled.div
  id="metrics-preference"
  onClick={[Function]}
>
  <FormattedMessage
    defaultMessage="Metrics"
    id="app.containers.AgentPreferencesMenu.metrics"
    values={Object {}}
  />
  <Styled(SimpleCaretIconSVG)
    direction="right"
    open={false}
    size={10}
  />
</styled.div>
`;
